src/seedu/addressbook/commands/SortCommand.java

```diff
@@ -0,0 +1,36 @@
+package seedu.addressbook.commands;
+
+import seedu.addressbook.data.person.ReadOnlyPerson;
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.ArrayList;
+
+
+/**
+ * Sorts all persons in the address book alphabetically and lists to the user.
+ */
+public class SortCommand extends Command {
+
+    public static final String COMMAND_WORD = "sort";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Displays all persons in the address book alphabetically as a list with index numbers.\n"
+            + "Example: " + COMMAND_WORD;
+
+
+    @Override
+    public CommandResult execute() {
+        List<ReadOnlyPerson> allPersons = new ArrayList<>(addressBook.getAllPersons().immutableListView());
+        Comparator<ReadOnlyPerson> nameComparator = new Comparator<ReadOnlyPerson>() {
```

Comment:
Combining class definition and object creation is not a recommended practice. Can you think of a reason why ?